packages/patternfly-4/react-core/src/components/Select/Select.tsx

```diff
@@ -440,6 +484,22 @@ class Select extends React.Component<SelectProps & InjectedOuiaProps, SelectStat
                     </div>
                   )}
                 </div>
+                {hasInlineFilter &&
+                  onClear !== Select.defaultProps.onClear &&
+                  (selections && (Array.isArray(selections) && selections.length > 0)) && (
+                    <button
+                      className={css(buttonStyles.button, buttonStyles.modifiers.plain, styles.selectToggleClear)}
+                      onClick={e => {
+                        this.clearSelection(e);
+                        onClear(e);
+                      }}
+                      aria-label={ariaLabelClear}
+                      type="button"
+                      disabled={isDisabled}
+                    >
+                      <TimesCircleIcon aria-hidden />
+                    </button>
+                  )}
```

Comment:
Currently, the close button is only added when using the `typeahead` feature. Adding it for `hasInlineFilter` seems inconsistent with the remaining examples?

IMO, I feel that the clear button shouldn't be limited to `typeahead` and `hasInlineFilter`. Seems like this feature would be useful, regardless? 

Can we show the clear button if `onClear` is present?